### PR TITLE
allow heli RSC on a different channel

### PR DESCRIPTION
**Pull request by tridge**
this makes it possible to have the heli RSC on a different channel

this is needed for the XRacer which only has 6 PWM outputs

### CODE REVIEW

File: libraries/AP_Motors/AP_MotorsHeli_RSC.cpp [hunk 1]
```diff
@@ -174,5 +174,6 @@ void AP_MotorsHeli_RSC::write_rsc(int16_t servo_out)
         _servo_output.calc_pwm();
 
         hal.rcout->write(_servo_output_channel, _servo_output.radio_out);
+        RC_Channel_aux::set_servo_out(RC_Channel_aux::k_heli_rsc, servo_out);
```
Comment (fhedberg) [inline libraries/AP_Motors/AP_MotorsHeli_RSC.cpp]: I think RC_Channel_aux::k_heli_rsc should be passed into AP_MotorsHeli_RSC just like _servo_output_channel. There could be multiple RSC instances, for example if you're using a direct-drive tail.

**Comment by fhedberg** [comment]
This is great - I'm all for using AUX function assignment for all output channels in AP_Motors.

**Comment by R-Lefebvre** [comment]
I totally support allowing users to move the RSC output around. However, I think that we should make this a complete solution instead of optionally twinning the output as done here.

I agree with Frederick's comments, that all of the outputs should be assignable.  Including motor outputs on multirotors.

Reviewed.

I approve of this change as long as we accept that this is a temporary fix and might change again at some point in the future.

**Comment by rmackay9** [comment]
But doesn't this mean that the rsc output appears on two channels?  RC8 (output) and whatever channel has it's FUNCTION set to RC_Channel_aux::k_heli_rsc?  That seems pretty strange to me and I can already hear the support question, "I moved the heli main rotor to RC6 but my gimbal won't work even though it's connected to RC8)".

**Comment by fhedberg** [comment]
Yeah... I played around with it being optional to use AUX functions a while back for https://github.com/diydrones/ardupilot/pull/2914 with https://github.com/fhedberg/ardupilot/commit/da1c73f14d293cc5105496459e4b8f1f6ac20edd, but I would much rather see it be default and have the AUX function assignments with per-frame parameter defaults to mimic the current behaviour.

**Comment by tridge** [comment]
we could add a RC_Channel_aux::set_servo_out_default() which takes a default channel. It would only send to the default channel if there is no function of that type assigned.
That would retain backwards compatibility as well as allow channel assignment.

**Comment by R-Lefebvre** [comment]
Agree with you Randy.  But this is just a temporary fix.  The correct fix is exactly as Frederick suggests.

It's the same thing as the Interlock input for Helis.
Currently it forced onto Ch8, and should be flexible.

Tridge, seems OK.  I would prefer to just have helis take Ch8 as a default.  Simple as that.  If somebody unassigns Ch8 and doesn't reassign something else, they won't be able to get the motor to start, but, they should read the Wiki then.

**Comment by tridge** [comment]
I have reworked the PR using a new RC_Channel_aux::set_aux_channel_default() API
This allows Heli to specify a default channel, but the user can override it by setting a function on a channel
If the default is used then the parameters in any logs will reflect the channel used

**Comment by rmackay9** [comment]
Sorry to be annoying but can I ask that this be rebased on master?  I could do it but I think it's safer if the author (in this case Tridge) does it.

**Comment by rmackay9** [comment]
I think the _servo_output object is never used with this change so we could/should remove it as a member of the AP_MotorsHeli_RSC class.

**Comment by rmackay9** [comment]
I've added the need-rework but that's just so it's more clear that it needs to be rebased.  I'm happy to take care of the other clean-up bits that i've mentioned above.

**Comment by tridge** [comment]
should be all fixed up now, thanks!

**Comment by tridge** [comment]
pushed after discussion with Randy

**Pull request merged**